python-package/lightgbm/sklearn.py

```diff
@@ -451,34 +426,26 @@ def apply(self, X, num_iteration=0):
         -------
         X_leaves : array_like, shape=[n_samples, n_trees]
         """
-        return self._Booster.predict(X,
-                                     pred_leaf=True,
-                                     num_iteration=num_iteration)
+        return self.booster_.predict(X, pred_leaf=True, num_iteration=num_iteration)
 
-    def evals_result(self):
-        """
-        Return the evaluation results.
-
-        Returns
-        -------
-        evals_result : dictionary
-        """
-        if self.evals_result_:
-            evals_result = self.evals_result_
-        else:
-            raise LightGBMError('No results found.')
-
-        return evals_result
-
-    def feature_importance(self):
-        """
-        Feature importances
+    @property
+    def booster_(self):
```

Comment:
@guolinke no, just sklearn names attributes with an underline suffix, see http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html#sklearn.ensemble.GradientBoostingRegressor

@henry0312 you cannot set property either. `@booster_.setter` is a setter. self._Booster should be private but it won't be in python, so `gbm._Booster = another_Booster` won't raise Exception but `gbm.booster_ = another_Booster` will.
I think xgboost may not consider this question in their early stage, and it will be painful for them to change api now.